### PR TITLE
refactor(runtime): using builder for mysql server

### PR DESCRIPTION
**Pull request by mlycore**
### What is changed and how it works?
- Using `MySqlServerBuilder` for `MySqlServer`

### CODE REVIEW

File: pisa-proxy/protocol/mysql/src/server/packet.rs [hunk 1]
```diff
@@ -14,11 +14,11 @@
 
 use std::{
     io::{Error, ErrorKind},
-    ptr::copy_nonoverlapping,
+    ptr::copy_nonoverlapping, u8,
```
Comment (wbtlb) [inline pisa-proxy/protocol/mysql/src/server/packet.rs:17]: if `u8` and `net::TcpStream` is not used, remove it.

File: pisa-proxy/protocol/mysql/src/server/conn.rs [hunk 2]
```diff
@@ -49,6 +49,7 @@ const DEFAULT_CAPABILITY: u32 = CLIENT_LONG_PASSWORD
     | CLIENT_PLUGIN_AUTH_LENENC_CLIENT_DATA;
 
 // client <=> Pisa
+
```
Comment (wbtlb) [inline pisa-proxy/protocol/mysql/src/server/conn.rs:52]: remove null line

File: pisa-proxy/runtime/mysql/src/server/server.rs [hunk 3]
```diff
@@ -50,42 +54,109 @@ pub struct MySqlServer {
...
-        client: TcpStream,
-        pool: Pool<ClientConn>,
-        lb: Arc<Mutex<BalanceType>>,
-        proxy_config: ProxyConfig,
-        parser: Arc<Parser>,
-        ast_cache: Arc<plMutex<ParserAstCache>>,
+impl MySqlServerBuilder {
+    pub fn new(
+        socket: TcpStream,
+        lb: Arc<Mutex<Box<dyn LoadBalance + Send + Sync>>>,
```
Comment (wbtlb) [inline pisa-proxy/runtime/mysql/src/server/server.rs:77]: lb has been changed to `enum`, update branch first

File: pisa-proxy/runtime/mysql/src/mysql.rs [hunk 4]
```diff
@@ -14,9 +14,11 @@
 
 use std::sync::Arc;
 
+use bytes::BytesMut;
```
Comment (wbtlb) [inline pisa-proxy/runtime/mysql/src/mysql.rs:17]: remove unused packet

File: pisa-proxy/protocol/mysql/src/server/packet.rs (hunk 1 above)
Comment (mlycore) [reply pisa-proxy/protocol/mysql/src/server/packet.rs:17]: fixed

File: pisa-proxy/runtime/mysql/src/server/server.rs (hunk 3 above)
Comment (mlycore) [reply pisa-proxy/runtime/mysql/src/server/server.rs]: fixed

File: pisa-proxy/protocol/mysql/src/server/conn.rs (hunk 2 above)
Comment (mlycore) [reply pisa-proxy/protocol/mysql/src/server/conn.rs:52]: fixed

File: pisa-proxy/runtime/mysql/src/mysql.rs (hunk 4 above)
Comment (mlycore) [reply pisa-proxy/runtime/mysql/src/mysql.rs:17]: fixed

**Review by wbtlb — APPROVED** [review]

**Pull request merged**